ext/dogstatsd_client.c

```diff
@@ -70,7 +70,12 @@ void ddtrace_dogstatsd_client_rinit(void) {
             host = url;
             port = NULL;
         } else {
-            host = ZSTR_VAL(get_DD_AGENT_HOST());
+            zend_string *dogstatsd_host = get_DD_DOGSTATSD_HOST();
+            if (ZSTR_LEN(dogstatsd_host) > 0) {
+                host = ZSTR_VAL(dogstatsd_host);
+            } else {
+                host = ZSTR_VAL(get_DD_AGENT_HOST());
+            }
```

Comment:
This code should rather be refactored to use ddtrace_dogstatsd_url()... And this host being also checked there then.

Reply:
Good point. I was going for a full integration test but that seems like overkill here. I just need to verify that the new configuration is being read.

Reply:
You're conflating the comments - ddtrace_dogstatsd_url is a C function, I wasn't talking about the test here.
This url is not exposed directly to the PHP code.

Reply:
I don't think you can easily.
Also, if you want to test against not-localhost, you might want to bind a socket on e.g. 172.0.0.2 instead.